modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientPrimaryReplicaTracker.java

```diff
@@ -81,6 +86,10 @@ public class ClientPrimaryReplicaTracker implements EventListener<EventParameter
 
     private final SchemaSyncService schemaSyncService;
 
+    private final LowWatermark lowWatermark;
+
+    private final DeferredEventsQueue<DestroyTableEvent> deferredEventsQueue = new DeferredEventsQueue<>(DestroyTableEvent::catalogVersion);
```

Comment:
I rewrote a queue structure.
Unfortunately, we can't use a timestamp here, because `activationTime` is internal Catalog thing,
but we can track catalogVersion, and process deferred destruction events, when catalogVersions dive under the LWM.